server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -406,14 +415,27 @@ private void setLastRunTimeMillis() {
         lastRunTimeMillis.getAndUpdate(l -> Math.max(l, currentTimeMillisSupplier.getAsLong()));
     }
 
+    protected void setIndexCreateBlock(final ActionListener<Void> listener, boolean indexCreateBlock) {
+        final ActionListener<Void> wrappedListener = ActionListener.wrap(r -> { listener.onResponse(r); }, e -> {
+            logger.debug("setting index create block failed", e);
+            listener.onFailure(e);
+        });
+
+        final Settings indexCreateBlockSetting = indexCreateBlock
+            ? Settings.builder().put(Metadata.SETTING_CREATE_INDEX_BLOCK_SETTING.getKey(), Boolean.TRUE.toString()).build()
+            : Settings.builder().putNull(Metadata.SETTING_CREATE_INDEX_BLOCK_SETTING.getKey()).build();
+
+        client.admin()
+            .cluster()
+            .prepareUpdateSettings()
+            .setPersistentSettings(indexCreateBlockSetting)
+            .execute(ActionListener.map(wrappedListener, r -> null));
+    }
+
     protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener<Void> listener, boolean readOnly) {
         // set read-only block but don't block on the response
-        ActionListener<Void> wrappedListener = ActionListener.wrap(r -> {
-            setLastRunTimeMillis();
-            listener.onResponse(r);
-        }, e -> {
+        ActionListener<Void> wrappedListener = ActionListener.wrap(r -> { listener.onResponse(r); }, e -> {
```

Comment:
for my understanding y did we remove last run time?

Reply:
As commented by Gaurav [here](https://github.com/opensearch-project/OpenSearch/pull/5852#discussion_r1080975729), ```setLastRunTimeMillis()``` is only meant for reroute operations as it needs to honor ```diskThresholdSettings.getRerouteInterval().millis()```.

Reply:
If we remove `setLastRunTimeMillis ` it will try to reroute once even though we don't need to reroute.

Reply:
Addressed the comment